### ISSUE TITLE
Warning message

### ISSUE THREAD

**Issue by lseufer** [twrecked/hass-aarlo#154]
Here is a warning message I'm getting:

general exception can only concatenate str (not "NoneType") to str in pyaarlo/__init__.py (WARNING)

**Comment by twrecked** [comment]
Is the from `home-assistant.log`, if not do you mind looking in there (it's in your `/config` directory)? There could be more information near the exception that would allow me to narrow down my search in the init code.

Also, can you restart the system with some extra debug enabled? This extra debug will go `home-assistant.log` and will give me an even better chance of finding the problem.

```yaml
logger:
  default: info
  logs:
    pyaarlo: debug
```

**Comment by lseufer** [comment]
Here you go. I hope I am not pasting sensitive data tho.
`2020-01-23 18:45:47 DEBUG (ArloBackgroundWorker) [pyaarlo] set:59U1817LAC956/lastImageSource=capture/01-22 09:54
2020-01-23 18:45:47 DEBUG (ArloBackgroundWorker) [pyaarlo] set:59U1817LAC956/lastCapture=01-22 09:54
2020-01-23 18:45:47 DEBUG (ArloBackgroundWorker) [pyaarlo] set:59U1817LAC956/presignedLastImageData=b'\xff\xd8\xff\xfe\x00\x10Lavc58.35.100
2020-01-23 18:45:47 DEBUG (ArloBackgroundWorker) [pyaarlo] set:59U1817LAC956/activityState=idle
2020-01-23 18:45:47 DEBUG (ArloBackgroundWorker) [pyaarlo] getting image for Sonnette
2020-01-23 18:45:47 DEBUG (ArloBackgroundWorker) [pyaarlo] set:A2G19BKBA0633/lastImageSource=capture/01-22 08:51
2020-01-23 18:45:47 DEBUG (ArloBackgroundWorker) [pyaarlo] set:A2G19BKBA0633/lastCapture=01-22 08:51
2020-01-23 18:45:47 DEBUG (ArloBackgroundWorker) [pyaarlo] set:A2G19BKBA0633/presignedLastImageData=b'\xff\xd8\xff\xe0\x00\x10JFIF\x00\x01\
2020-01-23 18:45:47 DEBUG (ArloBackgroundWorker) [pyaarlo] set:A2G19BKBA0633/activityState=idle
2020-01-23 18:45:47 DEBUG (ArloBackgroundWorker) [pyaarlo] reloading cache for Camera du jardin
2020-01-23 18:45:47 DEBUG (ArloBackgroundWorker) [pyaarlo] set:52M1817GBD960/capturedToday=0
2020-01-23 18:45:47 DEBUG (ArloBackgroundWorker) [pyaarlo] reloading cache for Camera de la piscine
2020-01-23 18:45:47 DEBUG (ArloBackgroundWorker) [pyaarlo] set:59U17B7LD9AFE/capturedToday=0
2020-01-23 18:45:47 DEBUG (ArloBackgroundWorker) [pyaarlo] reloading cache for Camera du garage
2020-01-23 18:45:47 DEBUG (ArloBackgroundWorker) [pyaarlo] set:59U1817LAC956/capturedToday=0
2020-01-23 18:45:47 DEBUG (ArloBackgroundWorker) [pyaarlo] reloading cache for Sonnette
2020-01-23 18:45:47 DEBUG (ArloBackgroundWorker) [pyaarlo] set:A2G19BKBA0633/capturedToday=0
2020-01-23 18:45:47 DEBUG (ArloBackgroundWorker) [pyaarlo] initial refresh
2020-01-23 18:45:48 DEBUG (ArloBackgroundWorker) [pyaarlo] deduced v2 api
2020-01-23 18:45:48 DEBUG (ArloBackgroundWorker) [pyaarlo] mode0<=M=>disarmed
2020-01-23 18:45:48 DEBUG (ArloBackgroundWorker) [pyaarlo] set:A2G19BKBA0633/modeIdToName/mode0=disarmed
2020-01-23 18:45:48 DEBUG (ArloBackgroundWorker) [pyaarlo] set:A2G19BKBA0633/modeNameToId/disarmed=mode0
2020-01-23 18:45:48 DEBUG (ArloBackgroundWorker) [pyaarlo] set:A2G19BKBA0633/modeIsSchedule/disarmed=False
2020-01-23 18:45:48 DEBUG (ArloBackgroundWorker) [pyaarlo] mode1<=M=>armed
2020-01-23 18:45:48 DEBUG (ArloBackgroundWorker) [pyaarlo] set:A2G19BKBA0633/modeIdToName/mode1=armed
2020-01-23 18:45:48 DEBUG (ArloBackgroundWorker) [pyaarlo] set:A2G19BKBA0633/modeNameToId/armed=mode1
2020-01-23 18:45:48 DEBUG (ArloBackgroundWorker) [pyaarlo] set:A2G19BKBA0633/modeIsSchedule/armed=False
2020-01-23 18:45:48 DEBUG (ArloBackgroundWorker) [pyaarlo] schedule.1<=S=>Arm camera M-F 8-5
2020-01-23 18:45:48 DEBUG (ArloBackgroundWorker) [pyaarlo] set:A2G19BKBA0633/modeIdToName/schedule.1=Arm camera M-F 8-5
2020-01-23 18:45:48 DEBUG (ArloBackgroundWorker) [pyaarlo] set:A2G19BKBA0633/modeNameToId/arm camera m-f 8-5=schedule.1
2020-01-23 18:45:48 DEBUG (ArloBackgroundWorker) [pyaarlo] set:A2G19BKBA0633/modeIsSchedule/arm camera m-f 8-5=True
2020-01-23 18:45:48 DEBUG (ArloBackgroundWorker) [pyaarlo] set:A2G19BKBA0633/activeSchedule=Arm camera M-F 8-5
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] sending cameras to A2G19BKBA0633
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] Sonnette BASE got cameras
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] *DEVICE* Sonnette: got cameras event {'interfaceVersion': 1, 'serialNumber': 'A2G19BKBA0633', 'batteryLevel': 100, 'batteryTech': 'Rechargeable', 'chargerTech': 'VAC Charger', 'chargingState': 'Off', 'chargeOnly': False, 'signalStrength': 3, 'brightness': 0, 'mirror': False, 'flip': False, 'zoom': {'topleftx': 0, 'toplefty': 0, 'bottomrightx': 1536, 'bottomrighty': 1536}, 'mic': {'mute': False, 'volume': 100}, 'speaker': {'mute': False, 'volume': 100}, 'streamingMode': 'eventBased', 'continuousStreamState': 'inactive', 'motion': {'sensitivity': 5, 'zones': [{'name': 'Entrée', 'id': '68cfd942-a579-4e34-9146-9c3c8db6ba29', 'coords': [{'x': 0.181708, 'y': 0.135204}, {'x': 1, 'y': 0.135204}, {'x': 1, 'y': 0.992592}, {'x': 0.181708, 'y': 0.992592}], 'color': 8524960}]}, 'resolution': {'width': 1536, 'height': 1536}, 'idleLedEnable': True, 'privacyActive': False, 'standbyActive': False, 'connectionState': 'available', 'activityState': 'idle', 'swVersion': '1.14.1.0_368_ef57d3c_9d35dae', 'hwVersion': '1.2', 'modelId': 'AVD1001A', 'motionSetupModeEnabled': False, 'motionSetupModeSensitivity': 80, 'motionDetected': False, 'audioDetected': False, 'hasStreamed': True, 'olsonTimeZone': 'Europe/Belgrade', 'name': '', 'nightVisionMode': 1, 'videoMode': 'superWide', 'hdr': 'auto', 'updateAvailable': None, 'blockNotifications': {'block': False, 'duration': 0, 'endTime': 0}, 'bestLocalLiveStreaming': 'off', 'traditionalChime': False, 'traditionalChimeDuration': 320, 'traditionalChimeType': 'none', 'pirLedEnable': True, 'callLedEnable': True, 'liveFeed': True, 'silentMode': {}, 'chimes': {}, 'voiceMailEnabled': False, 'sipCallActive': False, 'states': {}}
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/batteryLevel=100
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/batteryTech=Rechargeable
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/chargerTech=VAC Charger
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/chargingState=Off
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/signalStrength=3
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/brightness=0
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/mirror=False
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/flip=False
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/speaker={'mute': False, 'volume': 100}
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/privacyActive=False
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/connectionState=available
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/activityState=idle
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/motionSetupModeEnabled=False
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/motionSetupModeSensitivity=80
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/motionDetected=False
2020-01-23 18:45:48 DEBUG (ArloBackgroundWorker) [pyaarlo] deduced v2 api
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/audioDetected=False
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] Sonnette CAMERA got one cameras
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] *DEVICE* Sonnette: got cameras event {'interfaceVersion': 1, 'serialNumber': 'A2G19BKBA0633', 'batteryLevel': 100, 'batteryTech': 'Rechargeable', 'chargerTech': 'VAC Charger', 'chargingState': 'Off', 'chargeOnly': False, 'signalStrength': 3, 'brightness': 0, 'mirror': False, 'flip': False, 'zoom': {'topleftx': 0, 'toplefty': 0, 'bottomrightx': 1536, 'bottomrighty': 1536}, 'mic': {'mute': False, 'volume': 100}, 'speaker': {'mute': False, 'volume': 100}, 'streamingMode': 'eventBased', 'continuousStreamState': 'inactive', 'motion': {'sensitivity': 5, 'zones': [{'name': 'Entrée', 'id': '68cfd942-a579-4e34-9146-9c3c8db6ba29', 'coords': [{'x': 0.181708, 'y': 0.135204}, {'x': 1, 'y': 0.135204}, {'x': 1, 'y': 0.992592}, {'x': 0.181708, 'y': 0.992592}], 'color': 8524960}]}, 'resolution': {'width': 1536, 'height': 1536}, 'idleLedEnable': True, 'privacyActive': False, 'standbyActive': False, 'connectionState': 'available', 'activityState': 'idle', 'swVersion': '1.14.1.0_368_ef57d3c_9d35dae', 'hwVersion': '1.2', 'modelId': 'AVD1001A', 'motionSetupModeEnabled': False, 'motionSetupModeSensitivity': 80, 'motionDetected': False, 'audioDetected': False, 'hasStreamed': True, 'olsonTimeZone': 'Europe/Belgrade', 'name': '', 'nightVisionMode': 1, 'videoMode': 'superWide', 'hdr': 'auto', 'updateAvailable': None, 'blockNotifications': {'block': False, 'duration': 0, 'endTime': 0}, 'bestLocalLiveStreaming': 'off', 'traditionalChime': False, 'traditionalChimeDuration': 320, 'traditionalChimeType': 'none', 'pirLedEnable': True, 'callLedEnable': True, 'liveFeed': True, 'silentMode': {}, 'chimes': {}, 'voiceMailEnabled': False, 'sipCallActive': False, 'states': {}}
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/batteryLevel=100
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/batteryTech=Rechargeable
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/chargerTech=VAC Charger
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/chargingState=Off
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/signalStrength=3
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/brightness=0
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/mirror=False
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/flip=False
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/speaker={'mute': False, 'volume': 100}
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/privacyActive=False
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/connectionState=available
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/activityState=idle
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/motionSetupModeEnabled=False
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/motionSetupModeSensitivity=80
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/motionDetected=False
2020-01-23 18:45:48 DEBUG (ArloEventStream) [pyaarlo] set:A2G19BKBA0633/audioDetected=False
2020-01-23 18:45:48 WARNING (ArloEventStream) [pyaarlo] general exception can only concatenate str (not "NoneType") to str
2020-01-23 18:45:49 DEBUG (ArloBackgroundWorker) [pyaarlo] mode0<=M=>disarmed
2020-01-23 18:45:49 DEBUG (ArloBackgroundWorker) [pyaarlo] set:4RD2777VA0948/modeIdToName/mode0=disarmed
2020-01-23 18:45:49 DEBUG (ArloBackgroundWorker) [pyaarlo] set:4RD2777VA0948/modeNameToId/disarmed=mode0
2020-01-23 18:45:49 DEBUG (ArloBackgroundWorker) [pyaarlo] set:4RD2777VA0948/modeIsSchedule/disarmed=False
2020-01-23 18:45:49 DEBUG (ArloBackgroundWorker) [pyaarlo] mode1<=M=>armed
2020-01-23 18:45:49 DEBUG (ArloBackgroundWorker) [pyaarlo] set:4RD2777VA0948/modeIdToName/mode1=armed
2020-01-23 18:45:49 DEBUG (ArloBackgroundWorker) [pyaarlo] set:4RD2777VA0948/modeNameToId/armed=mode1
2020-01-23 18:45:49 DEBUG (ArloBackgroundWorker) [pyaarlo] set:4RD2777VA0948/modeIsSchedule/armed=False
2020-01-23 18:45:49 DEBUG (ArloBackgroundWorker) [pyaarlo] schedule.1<=S=>schedule.1
2020-01-23 18:45:49 DEBUG (ArloBackgroundWorker) [pyaarlo] set:4RD2777VA0948/modeIdToName/schedule.1=schedule.1
2020-01-23 18:45:49 DEBUG (ArloBackgroundWorker) [pyaarlo] set:4RD2777VA0948/modeNameToId/schedule.1=schedule.1
2020-01-23 18:45:49 DEBUG (ArloBackgroundWorker) [pyaarlo] set:4RD2777VA0948/modeIsSchedule/schedule.1=True
2020-01-23 18:45:49 DEBUG (ArloBackgroundWorker) [pyaarlo] set:4RD2777VA0948/activeSchedule=schedule.1
`

**Comment by twrecked** [comment]
That's great. If you want to edit and delete that trace that's cool, I got a copy!

**Comment by lseufer** [comment]
Edit:  here it is but I don't use it for Arlo
alarm_control_panel:
  - platform: ifttt
    name: Somfy
    optimistic: true

**Comment by twrecked** [comment]
I tried removing my alarm panel and the warning didn't appear. On Arlo do you have any empty alarm mode names?

I'll probably need to get a copy of your event stream on start up but, before we try that, can you tell me what kind of cameras you are using? I have a very small subset here so that could be an issue.

**Comment by lseufer** [comment]
I have 3 Arlo Pro H11 and one video doorbell. 

I don't have any empty mode name on Arlo, but as the app is set up in French I do have some funky UTF-8 characters. Like in "Géolocalisation" which may cause the error now that I think of it.

**Comment by twrecked** [comment]
I'll try the name tonight. Does everything else seem to work? The cameras and doorbells appear?

**Comment by lseufer** [comment]
Everything else is peachy!

**Comment by baurandr** [comment]
I started getting this same message after I added my Arlo Video Doorbell.  It happens every 10 minutes.

**Comment by twrecked** [comment]
You still getting this after #198 ? It might have been the double device creation causing the issue.

**Comment by baurandr** [comment]
Yes, I was hoping the same thing but I am still getting this warning every 10 minutes.

**Comment by twrecked** [comment]
Every 10 minutes the code calls a `slow_refresh` function and this is obviously causing the problem. I think it's innocuous but I'd still like to fix it. Would either of you be willing to try the [0.6.89.beta4](https://github.com/twrecked/hass-aarlo/releases/tag/v0.6.89-beta.4) build? It should appear on HACS shortly.

What I need is the packets that come in around the time the error appears and the beta.4 release will print incoming packets in to the log.

Add the following to your `aarlo` config.

```yaml
aarlo:
  [...current config...]
  verbose_debug: True
```

And make sure logging is turned on:

```yaml
logger:
  default: info
  logs:
    pyaarlo: debug
```

I don't need the whole debug, what I'm looking for is packets that come in around the time the general exception appears. They will be marked with the phrase `packet in`. And if you're feeling like you have some extra time to kill (and to test the mechanism) you you can encrypt the data before pasting it here to hide it - see [here](https://github.com/twrecked/hass-aarlo#other-sensitive)

**Comment by baurandr** [comment]
Here is a redacted snippet of the `packet in`  immediately preceding the warning.  It is the same every time except the transaction ID changes as expected.  Let me know if this isn't the correct one or if I can help in any way.

```
2020-03-29 21:33:43 DEBUG (ArloEventStream) [pyaarlo] packet in={ 'action': 'is',
  'from': '<<deviceId redacted>>',
  'properties': [ { 'blockNotifications': { 'block': False,
                                            'duration': 0,
                                            'endTime': 0},
                    'callLedEnable': True,
                    'chimes': {},
                    'liveFeed': True,
                    'pirLedEnable': True,
                    'silentMode': {},
                    'sipCallActive': False,
                    'states': {},
                    'traditionalChime': True,
                    'traditionalChimeDuration': 10000,
                    'traditionalChimeType': 'digital',
                    'voiceMailEnabled': False}],
  'resource': 'doorbells',
  'to': '<<userId redacted>>',
  'transId': '<<transID redacted>>'}

2020-03-29 21:33:43 WARNING (ArloEventStream) [pyaarlo] general exception can only concatenate str (not "NoneType") to str
```

**Comment by twrecked** [comment]
That's perfect thanks. I reproduced the exception here first time!

**Comment by twrecked** [comment]
Fixed also by #207

**Comment by baurandr** [comment]
Confirmed here as well with 0.6.18.4.  No more warnings, thanks again!!!
